app/javascript/components/student/editor/TestRunSummary.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState, useEffect } from 'react'
+import { Submission, TestRunStatus } from '../Editor'
+import consumer from '../../../utils/action-cable-consumer'
+
+export type TestRun = {
+  submissionUuid: string
+  status: TestRunStatus
+  message: string
+  tests: Test[]
+}
+
+type Test = {
+  name: string
+  status: TestStatus
+  output: string
+}
+
+enum TestStatus {
+  PASS = 'pass',
+  FAIL = 'fail',
+}
+
+export function TestRunSummary({ submission }: { submission: Submission }) {
+  const [testRun, setTestRun] = useState<TestRun>({
+    submissionUuid: submission.uuid,
+    status: submission.testsStatus,
+    message: '',
+    tests: [],
+  })
+
+  useEffect(() => {
+    const subscription = consumer.subscriptions.create(
+      {
+        channel: 'Submission::TestRunsChannel',
+        submission_uuid: submission.uuid,
+      },
+      {
+        received: ({ test_run: testRun }: any) => {
+          setTestRun({
+            submissionUuid: testRun.submission_uuid,
+            status: testRun.status,
+            message: testRun.message,
+            tests: testRun.tests,
+          })
+        },
+      }
+    )
+
+    return () => {
+      subscription.unsubscribe()
+    }
+  }, [submission.uuid])
+
+  let content
```

Comment:
I'd extract this switch to a function and use return instead of assignment. Will make it easier if it gets more complex.